File: 783-search-in-a-binary-search-tree/search-in-a-binary-search-tree.java
class Solution {
    public TreeNode InOrder(TreeNode root, int val){
        if(root == null) return null;
        if(root.val == val) return root;


        TreeNode left = InOrder(root.left, val);
        TreeNode right = InOrder(root.right, val);

        if(left != null) return left; 

        return right;
 
    }
    public TreeNode searchBST(TreeNode root, int val) {
        return InOrder(root, val);
    }
}

File: 783-search-in-a-binary-search-tree/search-in-a-binary-search-tree.cpp
class Solution {
public:
    TreeNode* searchBST(TreeNode* root, int val) {
        while(root){
            if(root->val == val) return root;
            if(root->val > val) root = root->left;
            else root = root->right;
        }

        return NULL;
    }
};